src/main/java/com/minecolonies/coremod/placementhandlers/NamedGravePlacementHandler.java

```diff
@@ -0,0 +1,60 @@
+package com.minecolonies.coremod.placementhandlers;
+
+import com.ldtteam.structurize.placement.handlers.placement.IPlacementHandler;
+import com.ldtteam.structurize.util.PlacementSettings;
+import com.minecolonies.api.blocks.ModBlocks;
+import com.minecolonies.api.tileentities.TileEntityNamedGrave;
+import com.minecolonies.coremod.blocks.BlockMinecoloniesNamedGrave;
+import com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingGraveyard;
+import net.minecraft.block.BlockState;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.tileentity.ChestTileEntity;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class NamedGravePlacementHandler implements IPlacementHandler
+{
+    @Override
+    public boolean canHandle(@NotNull final World world, @NotNull final BlockPos pos, @NotNull final BlockState blockState)
+    {
+        return blockState.getBlock() instanceof BlockMinecoloniesNamedGrave;
+    }
+
+    @Override
+    public ActionProcessingResult handle(
+      @NotNull final World world,
+      @NotNull final BlockPos pos,
+      @NotNull final BlockState blockState,
+      @Nullable final CompoundNBT tileEntityData,
+      final boolean complete,
+      final BlockPos centerPos,
+      final PlacementSettings settings)
+    {
+        if (world.getBlockState(pos).getBlock() == ModBlocks.blockNamedGrave)
+        {
+            return ActionProcessingResult.SUCCESS;
+        }
+
+        BuildingGraveyard.handleBuildingNamedGrave(world, pos, blockState);
+        return ActionProcessingResult.SUCCESS;
+    }
+
+    @Override
+    public List<ItemStack> getRequiredItems(
+      @NotNull final World world,
+      @NotNull final BlockPos pos,
+      @NotNull final BlockState blockState,
+      @Nullable final CompoundNBT tileEntityData,
+      final boolean complete)
+    {
+        final List<ItemStack> itemList = new ArrayList<>();
```

Comment:
the named grave requires no items to be placed at all?

Reply:
Yes, it is free as it will not be placed by the builder while building a graveyard.

In a schematic: it is just a placeholder for the position and rotation of future NamedGraves that will be build on citizen death by the undertaker. 

As it's a deco block in the graveyard, the undertaker can "spawn" it out of thin air.